pcli/src/command/validator.rs

```diff
@@ -186,29 +194,39 @@ impl ValidatorCmd {
 
                 println!("Cast validator vote");
             }
-            ValidatorCmd::Definition(DefinitionCmd::Template { file }) => {
+            ValidatorCmd::Definition(DefinitionCmd::Template {
+                file,
+                tendermint_validator_keyfile,
+            }) => {
                 let (address, _dtk) = fvk.incoming().payment_address(0u64.into());
                 let identity_key = IdentityKey(fvk.spend_verification_key().clone());
                 // By default, the template sets the governance key to the same verification key as
                 // the identity key, but a validator can change this if they want to use different
                 // key material.
                 let governance_key = GovernanceKey(identity_key.0);
-                // Generate a random consensus key.
-                // TODO: not great because the private key is discarded here and this isn't obvious to the user
-                let consensus_key = ed25519_consensus::SigningKey::new(OsRng);
-
-                /* MAKESHIFT RAFT ZONE */
-                let signing_key_bytes = consensus_key.as_bytes().as_slice();
-                let verification_key_bytes = consensus_key.verification_key();
-                let verification_key_bytes = verification_key_bytes.as_bytes().as_slice();
-                // TODO(erwan): surely there's a better way to do this?
-                let keypair = [signing_key_bytes, verification_key_bytes].concat();
-                let keypair = keypair.as_slice();
-
-                let consensus_key = ed25519_dalek::Keypair::from_bytes(keypair).unwrap();
-                /* END */
 
-                let consensus_key = tendermint::PrivateKey::Ed25519(consensus_key).public_key();
+                // Honor the filepath to `priv_validator_key.json`, if set. Otherwise, generate
+                // a random pubkey and emit a warning about it.
+                let consensus_key: tendermint::PublicKey = match tendermint_validator_keyfile {
+                    Some(f) => {
+                        tracing::debug!(?f, "Reading tendermint validator pubkey from file");
+                        let tm_key_config: Value = serde_json::from_str(&std::fs::read_to_string(
+                            &f,
+                        )?)
+                        .context(format!(
+                            "Could not parse file as Tendermint validator config: {}",
+                            f
+                        ))?;
+                        serde_json::value::from_value::<tendermint::PublicKey>(
+                            tm_key_config["pub_key"].clone(),
+                        )
+                        .context(format!("Tendermint JSON file malformed: {}", f))?
```

Comment:
Not sure if this is worth documenting, but a nice property of this implementation is that it will ignore the private key component of the file, so it's still possible to use even if the JSON file only has the public key

Reply:
That's rad. I tested mostly with a completely wrong file, e.g. a non-json ascii text file, and confirmed the error message was displayed. That's enough to get someone to check the contents manually.